webapp/publisher/snaps/build_views.py

```diff
@@ -34,6 +35,13 @@ def get_builds(lp_snap, context, selection):
         lp_snap["builds_collection_link"][32:]
     )
 
+    if build_id:
```

Comment:
It will be better to create a new API call to launchpad to get only one build. It is not good to get all the builds and then select one.